integration test fixes for 4.18

DaanHoogland opened the pull request:
### Description

This PR...
<!--- Describe your changes in DETAIL - And how has behaviour functionally changed. -->

<!-- For new features, provide link to FS, dev ML discussion etc. -->
<!-- In case of bug fix, the expected and actual behaviours, steps to reproduce. -->

<!-- When "Fixes: #<id>" is specified, the issue/PR will automatically be closed when this PR gets merged -->
<!-- For addressing multiple issues/PRs, use multiple "Fixes: #<id>" -->
<!-- Fixes: # -->

<!--- ********************************************************************************* -->
<!--- NOTE: AUTOMATATION USES THE DESCRIPTIONS TO SET LABELS AND PRODUCE DOCUMENTATION. -->
<!--- PLEASE PUT AN 'X' in only **ONE** box -->
<!--- ********************************************************************************* -->

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

### Feature/Enhancement Scale or Bug Severity

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial


### Screenshots (if appropriate):


### How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, and the tests you ran to -->
<!-- see how your change affects other areas of the code, etc. -->


<!-- Please read the [CONTRIBUTING](https://github.com/apache/cloudstack/blob/main/CONTRIBUTING.md) document -->

DaanHoogland commented:
@blueorangutan package

blueorangutan commented:
@DaanHoogland a Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

DaanHoogland commented on test/integration/smoke/test_deploy_vm_extra_config_data.py:
```diff
@@ -63,34 +63,27 @@ def setUpClass(cls):
-
-        except Exception as e:
-            raise Exception("Warning: Exception during cleanup : %s" % e)
+        super(TestAddConfigtoDeployVM, cls).tearDownClass()
```
@weizhouapache, stolen from #7186. We are working on the same stuff ...
```suggestion
        super(TestAddConfigtoDeployVM, cls).tearDownClass()
        cls.reset_hosts_hugepages()
```

weizhouapache commented on test/integration/smoke/test_deploy_vm_extra_config_data.py:
```diff
@@ -130,6 +124,10 @@ def setUp(self):
         updateConfigurationResponse = self.apiclient.updateConfiguration(updateConfigurationCmd)
         self.debug("updated the parameter %s with value %s" % (
             updateConfigurationResponse.name, updateConfigurationResponse.value))
+        self.cleanup = []
```
it is empty by default, right ?

DaanHoogland replied:
yes, it is in this file only filled by deploy_vm()

weizhouapache reviewed (approved):
code lgtm

blueorangutan commented:
Packaging result: :heavy_check_mark: el7 :heavy_check_mark: el8 :heavy_check_mark: el9 :heavy_check_mark: debian :heavy_check_mark: suse15. SL-JID 5547

DaanHoogland commented:
@blueorangutan package

blueorangutan commented:
@DaanHoogland a Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

blueorangutan commented:
Packaging result: :heavy_check_mark: el7 :heavy_check_mark: el8 :heavy_check_mark: el9 :heavy_check_mark: debian :heavy_check_mark: suse15. SL-JID 5552

DaanHoogland commented:
@blueorangutan test matrix

blueorangutan commented:
@DaanHoogland a Trillian-Jenkins matrix job (centos7 mgmt + xenserver71, rocky8 mgmt + vmware67u3, centos7 mgmt + kvmcentos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian Build Failed (tid-6153)<b/>

blueorangutan commented:
<b>Trillian test result (tid-6156)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 44666 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr7190-t6156-kvm-centos7.zip
Smoke tests completed. 105 look OK, 2 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_02_deploy_vm_with_extraconfig_kvm | `Error` | 38.53 | test_deploy_vm_extra_config_data.py
test_03_update_vm_with_extraconfig_kvm | `Error` | 132.82 | test_deploy_vm_extra_config_data.py
test_01_deploy_vm_on_specific_host | `Error` | 37.09 | test_vm_deployment_planner.py
test_02_deploy_vm_on_specific_cluster | `Error` | 4.32 | test_vm_deployment_planner.py
test_03_deploy_vm_on_specific_pod | `Error` | 5.38 | test_vm_deployment_planner.py
test_04_deploy_vm_on_host_override_pod_and_cluster | `Error` | 4.37 | test_vm_deployment_planner.py
test_05_deploy_vm_on_cluster_override_pod | `Error` | 4.33 | test_vm_deployment_planner.py

blueorangutan commented:
<b>Trillian test result (tid-6154)</b>
Environment: xenserver-71 (x2), Advanced Networking with Mgmt server 7
Total time taken: 44798 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr7190-t6154-xenserver-71.zip
Smoke tests completed. 105 look OK, 2 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_07_deploy_vm_with_extraconfig_xenserver | `Error` | 46.71 | test_deploy_vm_extra_config_data.py
test_01_deploy_vm_on_specific_host | `Error` | 10.43 | test_vm_deployment_planner.py
test_02_deploy_vm_on_specific_cluster | `Error` | 9.41 | test_vm_deployment_planner.py
test_03_deploy_vm_on_specific_pod | `Error` | 9.40 | test_vm_deployment_planner.py
test_04_deploy_vm_on_host_override_pod_and_cluster | `Error` | 11.41 | test_vm_deployment_planner.py
test_05_deploy_vm_on_cluster_override_pod | `Error` | 10.46 | test_vm_deployment_planner.py

blueorangutan commented:
<b>Trillian test result (tid-6155)</b>
Environment: vmware-67u3 (x2), Advanced Networking with Mgmt server r8
Total time taken: 46820 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr7190-t6155-vmware-67u3.zip
Smoke tests completed. 105 look OK, 2 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
test_05_deploy_vm_with_extraconfig_vmware | `Error` | 4.40 | test_deploy_vm_extra_config_data.py
test_01_deploy_vm_on_specific_host | `Error` | 11.69 | test_vm_deployment_planner.py
test_02_deploy_vm_on_specific_cluster | `Error` | 10.69 | test_vm_deployment_planner.py
test_03_deploy_vm_on_specific_pod | `Error` | 10.66 | test_vm_deployment_planner.py
test_04_deploy_vm_on_host_override_pod_and_cluster | `Error` | 12.66 | test_vm_deployment_planner.py
test_05_deploy_vm_on_cluster_override_pod | `Error` | 13.67 | test_vm_deployment_planner.py

weizhouapache commented:
@DaanHoogland 
you need to create vm using VirtualMachine.create, or roll back all changes on deploy/destroy vm

DaanHoogland commented:
@blueorangutan package

blueorangutan commented:
@DaanHoogland a Jenkins job has been kicked to build packages. It will be bundled with  KVM, XenServer and VMware SystemVM templates. I'll keep you posted as I make progress.

DaanHoogland commented:
@weizhouapache I reverted changes on test_vm_deployment_planner as the failures seem intemitted, and I don't see a reason. I reverted most changes on test_deploy_vm_extra_config_data, but added the possibility to call on the marvin object directly.

blueorangutan commented:
Packaging result: :heavy_check_mark: el7 :heavy_check_mark: el8 :heavy_check_mark: el9 :heavy_check_mark: debian :heavy_check_mark: suse15. SL-JID 5570

weizhouapache commented:
> @weizhouapache I reverted changes on test_vm_deployment_planner as the failures seem intemitted, and I don't see a reason. I reverted most changes on test_deploy_vm_extra_config_data, but added the possibility to call on the marvin object directly.

The vm, which is created by deployVirtualMachine.deployVirtualMachineCmd, cannot be cleaned up by tearDown class.
need to use VirtualMachine.create, like other tests, then it can be cleaned up in tearDown.

DaanHoogland commented:
> > @weizhouapache I reverted changes on test_vm_deployment_planner as the failures seem intemitted, and I don't see a reason. I reverted most changes on test_deploy_vm_extra_config_data, but added the possibility to call on the marvin object directly.
> 
> The vm, which is created by deployVirtualMachine.deployVirtualMachineCmd, cannot be cleaned up by tearDown class. need to use VirtualMachine.create, like other tests, then it can be cleaned up in tearDown.

Yes, I realise that. I think that is a big change now, it got me some request signing errors trying. I want to merge as is.

nvazquez reviewed (approved):
Code LGTM, can we run a final round of tests?

DaanHoogland commented:
@blueorangutan test

blueorangutan commented:
@DaanHoogland a Trillian-Jenkins test job (centos7 mgmt + kvm-centos7) has been kicked to run smoke tests

blueorangutan commented:
<b>Trillian test result (tid-6175)</b>
Environment: kvm-centos7 (x2), Advanced Networking with Mgmt server 7
Total time taken: 41159 seconds
Marvin logs: https://github.com/blueorangutan/acs-prs/releases/download/trillian/pr7190-t6175-kvm-centos7.zip
Smoke tests completed. 108 look OK, 0 have errors, 0 did not run
Only failed and skipped tests results shown below:


Test | Result | Time (s) | Test File
--- | --- | --- | ---
